### DISCUSSION TITLE
Add "reveal in explorer" to browser item context menu

### DISCUSSION THREAD

**Discussion by Did-Mas** [pymeasure/pymeasure#1186] [Ideas]
I think a useful feature would be the ability to open a file in the system's file explorer. Implementing this feature should be fairly straightforward, with the main challenge being cross-platform compatibility.

Here is an example implementation of a function triggered by a context menu action (tested on Windows):
```Python
def reveal_in_explorer(path_to_file):
    path_to_file = os.path.normpath(path_to_file)
    current_os = platform.system()

    if current_os == "Windows":
        subprocess.Popen(["explorer", "/select,", path_to_file])
    elif current_os == "Darwin":
        subprocess.Popen(["open", "-R", path_to_file])
    elif current_os == "Linux":
        subprocess.Popen(["xdg-open", os.path.dirname(path_to_file)])
    else:
        raise OSError(f"Unsupported operating system: {current_os}")
```

**Comment by CasperSchippers** [comment]
I can see that being useful indeed.  Feel free to open a PR to add this functionality.

I'm not sure if you already know where to start: you'd have to add the method to the ManagedWindowBase (similar to [`open_file_externally`](https://github.com/pymeasure/pymeasure/blob/master/pymeasure/display/windows/managed_window.py#L439-L455)), and add it to the context-menu in the [`browser_item_menu`](https://github.com/pymeasure/pymeasure/blob/35122c3136fab37c4412bb209c602825b5d429e4/pymeasure/display/windows/managed_window.py#L301) method.

PS I noticed in the [`open_file_externally`](https://github.com/pymeasure/pymeasure/blob/master/pymeasure/display/windows/managed_window.py#L439-L455) that 'start' is used for windows, which also works for directories; I'm not  sure if there is a preference of using one over the other??

**Comment by Did-Mas** [comment]
Using 'start' on Windows with a path to a directory does open the folder, but it doesn’t select the file, whereas the proposed command does. 

I’ve added this function in a new branch on my forked repository. Link to the commit: 8c4e82364988b4ab85b556e91929f1ee20f56076. I don’t currently have a way to test if it works on macOS and Linux. Should I go ahead and open a PR regardless, or would it be better to wait for confirmation on those platforms?

**Reply by CasperSchippers** [reply]
Ahh that is good to know; then your propose solution is definately better.

I'd suggest opening a PR regardless; I can test it on linux (although I probably won't have time before comming weekend). For macOS I we'll asume for now it'll work and hope someone raises an issue if it doesn't. I think we can't let a feature be blocked completely by not having access to macOS; best thing we can do is do our best to make a decent implementation and hope for the best, maybe we could add a comment stating that this has not been tested.

**Reply by Did-Mas** [reply]
Opened a PR:  #1188
